### ISSUE TITLE
Add option to download all email

### ISSUE THREAD

**Issue by mbilker** [nylas/N1#1198]
I am occasionaly offline, but still like to view my messages, even if the images fail to load.

I request an `NylasEnv.config` option be implemented to prevent the cleaning of messages in the database that are still associated with certain folders.

I prefer the default folder the messages are associated with is All Mail for Gmail and other platforms that follow this methodology and Inbox for other providers.

**Comment by bengotow** [comment]
Hey @mbilker — I think the issue is that we need to download more mail data. RIght now we only grab the thread and message metadata, and not the message bodies. To fully view your email offline, we'd need all the bodies as well. Is that what you're looking for?

**Comment by mbilker** [comment]
Yes, that is what I am looking for @bengotow.

**Comment by mbilker** [comment]
Changed the title to reflect ideas from #1291

**Comment by behrangsa** [comment]
Just in case, here was my bug report in #1291:

> On mobile email clients, as well as some desktop clients such as the now dead Mailbox, the entire mailbox which can sometimes be 10GB or more does not get downloaded.
> 
> Instead only the Inbox folder is downloaded. Other folders, when you click on them, download the most 50 recent emails on demand and as you scroll to the bottom of the view, the next 50 are downloaded, etc.
> 
> Add an option to N1 to make it work like this.

**Comment by behrangsa** [comment]
@mbilker When I tried N1 a few months ago, it downloaded my the entire Gmail mailbox. Does it still work the same way? This bug reports and the comments below it are a bit confusing.

**Comment by mbilker** [comment]
I agree with your points. I am working off of a mobile hotspot that sometimes works, but the connection may timeout or completely drop depending on the phone's connection algorithm and thermals.

**Comment by mbilker** [comment]
@behrangsa It does not download the whole mailbox to the client, but the Sync Engine stores a compressed copy of your mailbox. It downloads the most recent messages to a certain amount, but removes the rest from the client cache.

**Comment by behrangsa** [comment]
@mbilker Sorry I am a little bit confused. When a Gmail account is added for the first time to N1, does the Sync Engine download the entire mailbox and _then_ compress it, etc.?

Or does it only download the most recent messages?

Also is it possible to write a custom Sync Engine plugin?

**Comment by mbilker** [comment]
@behrangsa The Sync Engine is separate from N1. It is possible to write plugins for your **own**  instance of the Sync Engine.

**Comment by pizzaops** [comment]
When viewing this ticket you end up in a circular loop of tickets closed as dupes of others but eventually they all lead back here :smile:

**Comment by behrangsa** [comment]
What is Sync Engine by the way? Is it a server-side service?

**Comment by bengotow** [comment]
(Yep — http://github.com/nylas/sync-engine, it's a python service that can sync mail from almost any email provider.)

**Comment by bengotow** [comment]
Another user has requested this via ZenDesk: "I need to... be able to control how much it cached (e.g. save 1 year of mail offline)" https://nylas.zendesk.com/agent/tickets/4306

**Comment by pizzaops** [comment]
I assume this was meant to be an internal comment?

**Comment by bengotow** [comment]
Another user has requested this via ZenDesk: " I don't really need them all [2.2GB] cached locally. "

**Comment by diegogonzalez** [comment]
After all how is the behavior of the nylas email sync? It synchronize all email? I have 12 gb of mail, and I don't want it on my computer.

**Comment by ialexryan** [comment]
Hi there - I'm a new user and just installed Nylas Mail (Basic) and I'm noticing that my whole email history (10s of GBs) is currently being downloaded to my small boot SSD.

Can we please have an option to only download the most recent 3mo/6mo/1yr of messages? Or failing that, at least an option to change the save location would be very nice. Thanks!
